hazelcast/test/src/TestHelperFunctions.h

```diff
@@ -51,6 +51,11 @@
 #define WAIT_EQ_EVENTUALLY(expected, expression) WAIT_TRUE_EVENTUALLY((expected) == (expression))
 #define WAIT_NE_EVENTUALLY(expected, expression) WAIT_NE_EVENTUALLY((expected) != (expression))
 
+#define ASSERT_EQ_PTR(expected, actual, type) do{ \
```

Comment:
why do you use `do{}while(0)` here? Isn't sufficient just have `{}` here to create visibility scope (if it was the goal of course :) ) ?

Reply:
A common question. Here is the answer 
https://stackoverflow.com/a/257425